### ISSUE TITLE
Using duplicate job_id

### ISSUE THREAD

**Issue by SlimBeji** [rq/rq#1521]
Hello,

I noticed using duplicate job_id leads to some unexpected results.

```
@rq.job("low")
def test(input):
    from time import sleep
    print(input)
    sleep(2)
    print("dummy job finished")

test.queue("first execution", job_id="test")
test.queue("second execution", job_id="test", depends_on=["test"])
```

In the above example, it is expected to have "first_execution" and "second_execution" printed out however, "first execution" is printed out twice

**Comment by selwin** [comment]
RQ currently doesn't handle duplicated job_id so you should make sure that you job IDs are unique. RQ itself uses UUID so job IDs are always unique.

Having said that, I'm open to accepting a feature that checks for duplicate job IDs when custom job ID is supplied.

**Comment by aehlke** [comment]
It would be cool to have this work so that an attempt to insert duplicate tasks can behave idempotently

**Comment by selwin** [comment]
Agreed, this would be a nice feature to have. If someone wants to take a crack at this, I have some ideas on how this can be implemented.
